scipy/signal/tests/test_peak_finding.py

```diff
@@ -90,15 +85,15 @@ def test_empty(self):
         """Test with empty signal."""
         x = np.array([], dtype=np.float64)
         for array in _local_maxima_1d(x):
-            assert_equal(array, np.array([]))
-            assert_(array.base is None)
+            xp_assert_equal(array, np.array([], dtype=array.dtype))
```

Comment:
note to self (and below)

Reply:
So basically the issue here is the recommended way to test for an empty array. What would your recommendation be?

Reply:
I think we should just use `dtype=np.float64` here, assuming the function preserves dtype? `dtype=array.dtype` doesn't guard against regressions in the returned dtype at all.

Reply:
> assuming the function preserves dtype?

So in effect the suggestion is to tighten tests for the whole module. Where the dtypes have never been tested before. Wow.